icu4j/main/classes/core/src/com/ibm/icu/impl/units/SingleUnitImpl.java

```diff
@@ -0,0 +1,157 @@
+// © 2020 and later: Unicode, Inc. and others.
+// License & terms of use: http://www.unicode.org/copyright.html
+
+
+package com.ibm.icu.impl.units;
+
+import com.ibm.icu.util.MeasureUnit;
+
+public class SingleUnitImpl {
+    /**
+     * Simple unit index, unique for every simple unit, -1 for the dimensionless
+     * unit. This is an index into a string list in unit.txt {ConversionUnits}.
+     * <p>
+     * The default value is -1, meaning the dimensionless unit:
+     * isDimensionless() will return true, until index is changed.
+     */
+    private int index = -1;
+    /**
+     * SimpleUnit is the simplest form of a Unit. For example, for "square-millimeter", the simple unit would be "meter"Ò
+     * <p>
+     * The default value is "", meaning the dimensionless unit:
+     * isDimensionless() will return true, until index is changed.
+     */
+    private String simpleUnit = "";
+    /**
+     * Determine the power of the `SingleUnit`. For example, for "square-meter", the dimensionality will be `2`.
+     * <p>
+     * NOTE:
+     * Default dimensionality is 1.
+     */
+    private int dimensionality = 1;
+    /**
+     * SI Prefix
+     */
+    private MeasureUnit.SIPrefix siPrefix = MeasureUnit.SIPrefix.ONE;
+
+    public SingleUnitImpl clone() {
+        SingleUnitImpl result = new SingleUnitImpl();
+        result.index = this.index;
+        result.dimensionality = this.dimensionality;
+        result.simpleUnit = this.simpleUnit;
+        result.siPrefix = this.siPrefix;
+
+        return result;
+    }
+
+    public MeasureUnit build() {
+        MeasureUnitImpl measureUnit = new MeasureUnitImpl(this);
+        return measureUnit.build();
+    }
+
+    /**
+     * Generates an neutral identifier string for a single unit which means we do not include the dimension signal.
+     *
+     * @throws IllegalArgumentException
+     */
+    public String getNeutralIdentifier() {
+        StringBuilder result = new StringBuilder();
+        int posPower = Math.abs(this.getDimensionality());
+
+        assert posPower > 0 : "getIdentifier does not support the dimensionless";
+
+        if (posPower == 1) {
+            // no-op
+        } else if (posPower == 2) {
+            result.append("square-");
+        } else if (posPower == 3) {
+            result.append("cubic-");
+        } else if (posPower <= 15) {
+            result.append("pow");
+            result.append(posPower);
+            result.append('-');
+        } else {
+            throw new IllegalArgumentException("Unit Identifier Syntax Error");
+        }
+
+        result.append(this.getSiPrefix().getIdentifier());
+        result.append(this.getSimpleUnit());
+
+        return result.toString();
+    }
+
+    /**
+     * Compare this SingleUnitImpl to another SingleUnitImpl for the sake of
```

Comment:
Why not support the standard "Comparable" interface?

Reply:
Having realized that "compareTo" is not actually the same as "equals", it's probably not the best name (it sounds like it should behave like the standard Comparator, but doesn't).

When you have a partial comparator like this, a better approach is to have a method which returns an inner class Comparator<T> as a view over the fields.